### ISSUE TITLE
Target reported unsupported offsets

### ISSUE THREAD

**Issue by dirkwhoffmann** [bebbo/bgdbserver#4]
Hi Stefan,

I did some experiments with `bgdbserver` and got an error message which I don’t understand. This is what I did:

- I compiled a small C program using `m68k-amigaos-gcc`.
- I copied the program to an ADF together with `bgdbserver`.
- I launched FSUAE (Kickstart 2.0) and started `bgdbserver`:

![bgdb](https://user-images.githubusercontent.com/12561945/147056628-0074fe82-6baa-4cf4-81f4-82a434baae37.jpg)

- On the Mac side, I startet `m68k-amigaos-gdb`.
- I connected to port 8080 to start a debugging session.
- The debug session terminated with the following error message:

```
GNU gdb (GDB) 10.0.50.211207-144216-git
...
This GDB was configured as "--host=x86_64-apple-darwin21.1.0 --target=m68k-amigaos".
...
Reading symbols from fibo.exe...
(gdb) target remote :8080
Remote debugging using :8080
warning: Target reported unsupported offsets: Text=00c21348;Data=00c23f78;Bss=00c240f8
0x00c21348 in ?? ()
```

Is it a bug or am I doing something fundamentally wrong?

**Comment by bebbo** [comment]
that seems still ok.
what about using now:
```
b main
c
```
and then maybe
```
lay split
```

**Comment by dirkwhoffmann** [comment]
> what about using...

Yes, that worked. I can step through my program. Awesome!

<img width="682" alt="Bildschirmfoto 2021-12-22 um 13 46 15" src="https://user-images.githubusercontent.com/12561945/147095635-ad802e97-a16d-4cf0-9f21-313ed9e3a45e.png">

I've noticed that it reports a wrong contents for `fibo[1]`, most likely because `fibo` is a global variable. On your website you're stating "*local* variables and parameters are working now". 

Indeed, all local variables are reported correctly:

<img width="682" alt="Bildschirmfoto 2021-12-22 um 13 49 15" src="https://user-images.githubusercontent.com/12561945/147095940-fcacfa24-e542-4d9b-bdb4-0d72c583a0f3.png">

Background: I was trying to add a GDB server to my Amiga emulator (vAmiga) which turned out to be not so easy. At first, I canceled the project, but since I don't like failure, I am thinking of giving it another try. By using netcat I should be able to display the packages that are exchanged by `m68k-amigaos-gdb` on the Mac side and `bgdbserver` on the FS-UAE side. If I manage to mimic what your `bgdserver` is doing, it should be possible to connect `m68k-amigaos-gdb` directly to vAmiga.

**Comment by bebbo** [comment]
AFAIK: there are versions of fs-uae and winuae providing gdb support.

**Comment by dirkwhoffmann** [comment]
> AFAIK: there are versions of fs-uae and winuae providing gdb support.

Yes. The original plan was to make vAmiga compatible with the VSC plugin written by @prb28 in a similar way to what has already been achieved with UAE and FS-UAE. The project progress is documented in this thread: https://github.com/dirkwhoffmann/vAmiga/issues/618

At some point I put the project on hold. I stumbled over so many obstacles that I got the feeling that it will cost too much time at the end. However, I am still hoping that the integration will be achieved eventually. Right now, I think it's best to move forward in small steps. For me, the most promising step at the moment seems to be to make vAmiga mimic what your bgdbserver is doing.

**Comment by dirkwhoffmann** [comment]
Quick question: This is part of my recorded log (exchanged between `m68k-amigaos-gdb`and `bgdbserver`):

```
+$c#63
+$T051:00c21472;#22
+$p11#d2
+$00c21472#c3
```

I understand that in package `+$T051:xxxxxxxx;#yy`,  `xxxxxxxx` is the program counter, `05` is `SIGTRAP`, and `1` is the register number. Shouldn't it be `11` (decimal 17) instead of `1`? I guess you're transferring the PC to avoid the `p11` request that is coming immediately afterwards.  

BTW, a lot of conversation with vAmiga is already working. I can connect your `m68-amigaos-gdb` directly to my emulator and step through my test program (the values of variables are still wrong yet):

<img width="717" alt="Bildschirmfoto 2021-12-23 um 18 27 32" src="https://user-images.githubusercontent.com/12561945/147275087-03ee5bd8-ec5f-4cf8-b0f6-7c2a947ade33.png">

Emulator side:

<img width="988" alt="Bildschirmfoto 2021-12-23 um 18 42 03" src="https://user-images.githubusercontent.com/12561945/147275240-86e80bd4-769e-45ec-8ac3-4ee6828f2874.png">
